apps/charterafrica/src/lib/data/common/processPageDatasets.js

```diff
@@ -42,30 +50,77 @@ export default async function processPageDatasets(page, api, context) {
   const { blocks, breadcrumbs = [] } = page;
   const pageUrl = breadcrumbs[breadcrumbs.length - 1]?.url;
 
-  const { organizationId } = await api.findGlobal("openAfrica");
+  const datasetsIndex = blocks.findIndex(
+    ({ slug }) => slug === "datasetsAndDocuments"
+  );
 
-  const datasetsIndex = blocks.findIndex(({ slug }) => slug === "datasets");
+  if (datasetsIndex > -1) {
+    const {
+      showDatasets,
+      showDocuments,
+      labels: commonLabels,
+    } = blocks[datasetsIndex];
 
-  if (datasetsIndex > -1 && organizationId) {
-    const data = await fetchDatasets(organizationId, pageUrl, {
-      locale,
-    });
-    const { count, datasets, countries, tags, totalPages } = data;
+    if (showDatasets) {
+      const {
+        datasets: { organizationId, filterBar, labels },
+      } = blocks[datasetsIndex];
 
+      const data = await fetchDatasets(organizationId, pageUrl, {
+        locale,
+      });
+
+      blocks[datasetsIndex] = {
+        ...blocks[datasetsIndex],
+        datasets: {
+          data,
+          filterBar,
+          labels: {
+            ...commonLabels,
+            ...labels,
+          },
+          organizationId,
+        },
+      };
+    }
+
+    if (showDocuments) {
```

Comment:
As we discussed, documents and datasets need to be different pages (see https://pesayetu.pesacheck.org/data).

... an alternative single page implementation is possible (e.g. /data page) but one would have to manage what "tab" is current active between documents and datasets, etc.